src/cli.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import json
+import logging
+import re
+from typing import Any, Optional
+
+from ops import Container
+from ops.pebble import Error, ExecError
+from pydantic import AliasChoices, BaseModel, Field, field_serializer, field_validator
+
+from constants import ADMIN_PORT, CONFIG_FILE_NAME, DEFAULT_OAUTH_SCOPES, DEFAULT_RESPONSE_TYPES
+from exceptions import MigrationError
+
+logger = logging.getLogger(__name__)
+
+VERSION_REGEX = re.compile(r"Version:\s+(?P<version>v\d+\.\d+\.\d+)")
+
+
+class OAuthClient(BaseModel):
```

Comment:
The `OAuthClient` acts as the data interface between caller and oauth-client-related cli commands.